test/test_type_hints.py

```diff
@@ -118,13 +119,17 @@ def test_doc_examples(self):
                 )
             except OSError:
                 raise unittest.SkipTest('cannot symlink') from None
-            (stdout, stderr, result) = mypy.api.run([
-                '--cache-dir=.mypy_cache/doc',
-                '--no-strict-optional',  # needed because of torch.lu_unpack, see gh-36584
-                os.path.abspath(fn),
-            ])
+            repo_rootdir = Path(__file__).resolve().parent.parent
+            # TODO: Would be better not to chdir here, this affects the
+            # entire process!
+            with set_cwd(str(repo_rootdir)):
+                (stdout, stderr, result) = mypy.api.run([
+                    '--cache-dir=.mypy_cache/doc',
```

Comment:
It seems like it'd be cleaner to use the `--config-file` option and pass it an absolute path to the `mypy.ini` rather than changing the cwd, would that still work?

Reply:
that doesn't immediately work (it requires modifications to `mypy_plugins/check_mypy_version.py`), and I also simply prefer this because it matches up with the way we otherwise use `mypy` in CI (from the repo root)